docs/_tutorials/reference.md

```diff
@@ -79,6 +80,38 @@ The structure of the `payload` and `body` is detailed on the API site:
 ### Difference from listener middleware
 Listener middleware is used to implement logic across many listener functions (though usually not all of them). Listener middleware has the same arguments as the above listener functions, with one distinction: they also have a `next()` function that **must** be called in order to pass the chain of execution. Learn more about listener middleware [in the documentation](/bolt-js/concepts#listener-middleware).
 
+## Built-in listener middleware functions
+
+Bolt offers a variety of built-in listener middleware functions to help simplify development of your Slack applications. These middleware functions implement common patterns to help filter out or focus your own listener function implementations.
+
+These middleware functions are exported from the main `@slack/bolt` package for you to easily `import` in your applications:
+
+```javascript
+import { matchMessage } from '@slack/bolt';
+app.message(matchMessage('hello'), async ({ message, logger }) => {
+  // this function will now only execute if "hello" is present in the message
+});
+```
+
+A complete list of supported listener middleware functions follows.
+
+- `directMention()`: Filters out any message event whose text does not start with an @-mention of the handling app.
+- `ignoreSelf()`: Filters out any event that originates from the app.
```

Comment:
This middleware is designed to be used as a global middleware, and in fact, it is enabled as one of the built-in global custom middleware (developers can opt-out by passing `ignoreSelf: false` to the App constructor). We may want to categorize these into global or listener middleware.